sw/device/tests/consecutive_irqs/consecutive_irqs_test.c

```diff
@@ -0,0 +1,289 @@
+    uart_send_string("PLIC RX overflow trigger type set failed\n");
+    return false;
+  }
+  if (!dif_plic_irq_trigger_type_set(plic, kDifUartInterruptTxEmpty,
+                                     kDifPlicDisable)) {
+    uart_send_string("PLIC TX empty trigger type set failed\n");
+    return false;
+  }
+
+  // Set IRQ priorities to MAX
+  if (!dif_plic_irq_priority_set(plic, kDifPlicIrqIdUartRxOverflow,
+                                 PLIC_PRIORITY_MAX)) {
+    uart_send_string("PLIC priority set for RX overflow failed\n");
+    return false;
+  }
+  if (!dif_plic_irq_priority_set(plic, kDifPlicIrqIdUartTxOverflow,
+                                 PLIC_PRIORITY_MAX)) {
+    uart_send_string("PLIC priority set for TX empty failed\n");
+    return false;
+  }
+
+  // Set Ibex IRQ priority threshold level
+  if (!dif_plic_target_threshold_set(&plic0, PLIC_TARGET, PLIC_PRIORITY_MIN)) {
+    uart_send_string("PLIC threshold set failed\n");
+    return false;
+  }
+
+  // Enable IRQs in PLIC
+  if (!dif_plic_irq_enable_set(plic, kDifPlicIrqIdUartRxOverflow, PLIC_TARGET,
+                               kDifPlicEnable)) {
+    uart_send_string("PLIC Interrupt Enable for RX overflow failed\n");
+    return false;
+  }
+  if (!dif_plic_irq_enable_set(plic, kDifPlicIrqIdUartTxOverflow, PLIC_TARGET,
+                               kDifPlicEnable)) {
+    uart_send_string("PLIC Interrupt Enable for TX empty failed\n");
+    return false;
+  }
+
+  return true;
+}
+
+static bool execute_test(dif_uart_t *uart) {
+  // Force UART RX overflow interrupt, sleep for 10uS (should be more than
+  // enough for an IRQ to propogate), check if the IRQ has occured and has been
+  // handled appropriately, check that the IRQ has not been asserted more than
+  // once.
+  if (!dif_uart_irq_force(uart, kDifUartInterruptRxOverflow)) {
+    uart_send_string("TEST failed to force RX overflow IRQ\n");
+    return false;
+  }
+  usleep(10);
+  if (!uart_rx_overflow_handled) {
+    uart_send_string("TEST RX overflow IRQ has not been handled\n");
+    return false;
+  }
+  if (uart_handled_more_than_once) {
+    uart_send_string("TEST RX overflow IRQ was asserted more than once\n");
+    return false;
+  }
+
+  // Force UART TX empty interrupt, sleep for 10uS (should be more than enough
+  // for an IRQ to propogate), check if the IRQ has occured and has been handled
+  // appropriately, check that the IRQ has not been asserted more than once.
+  if (!dif_uart_irq_force(uart, kDifUartInterruptTxEmpty)) {
+    uart_send_string("TEST failed to force TX empty IRQ\n");
+    return false;
+  }
+  usleep(10);
+  if (!uart_tx_empty_handled) {
+    uart_send_string("TEST TX empty IRQ has not been handled\n");
+    return false;
+  }
+  if (uart_handled_more_than_once) {
+    uart_send_string("TEST TX empty IRQ was asserted more than once\n");
+    return false;
+  }
+
+  return true;
+}
+
+int main(int argc, char *argv[]) {
+  (void)argc;
+  (void)argv;
```

Comment:
You don't need to void unused function arguments.

Reply:
Done.